router/use-cases/check-auth.js

```diff
@@ -1,12 +1,13 @@
+const jwt = require('jsonwebtoken');
+
 class CheckAuth {
-  constructor({ allowedGroups, jwt }) {
+  constructor({ allowedGroups }) {
     this.allowedGroups = allowedGroups;
-    this.jwt = jwt;
   }
 
   execute({ token }) {
     try {
-      const payload = this.jwt.verify(token, process.env.JWT_SECRET);
+      const payload = jwt.verify(token, process.env.JWT_SECRET);
```

Comment:
Is it worth dependency injecting the JWT secret into the constructor so that this can be tested without any reliance on the environment?

Reply:
is there a problem with just set process.env.whatever in the test?

Reply:
No, I think this works fine functionally either way - just seems inconsistent with what we are doing for `PlanGateway` where we pass in the environment variable to the constructor. 

Depends how strict we are being on "clean" too I suppose, technically `process.env` is an external thing that should have an abstracted way of accessing it so we aren't tightly coupled to it.

https://github.com/LBHackney-IT/housing-needs-shared-plan/pull/48/files#diff-3cd7324760375b1497a3de7d3f1726a9R16